language/documentation/book/src/vector.md

```diff
@@ -38,42 +38,36 @@ assert(Vector::pop_back(&mut v) == 5, 42);
 
 ## Destroying and copying `vector`s
 
-Some behaviors of `vector<T>` depend on whether `T` is a resource type. For example, vectors containing resources cannot be implictly discarded like `v` in the example above--they must be explicitly destroyed with `Vector::destroy_empty`.
+Some behaviors of `vector<T>` depend on the abilities of the element type, `T`. For example, vectors containing elements that do not have `drop` cannot be implicitly discarded like `v` in the example above--they must be explicitly destroyed with `Vector::destroy_empty`.
 
 Note that `Vector::destroy_empty` will abort at runtime unless `vec` contains zero elements.
 
-```rust
-fun destroy_resource_vector<T: resource>(vec: vector<T>) {
+```move
+fun destroy_any_vector<T>(vec: vector<T>) {
     Vector::destroy_empty(vec) // deleting this line will cause a compiler error
 }
 ```
 
-This error would also happen for an unconstrained `T` as the type *might* be a resource.
+But no error would occur for dropping a vector that contains elements with `drop`
```

Comment:
Please keep some kind of punctuation, either period or colon, at the ends of sentences.

```suggestion
But no error would occur for dropping a vector that contains elements with `drop`.
```